src/abstractdataframe/reshape.jl

```diff
@@ -355,12 +364,52 @@ julia> unstack(df, :variable, :value, fill=0)
 ─────┼──────────────────────
    1 │ 1           1      2
    2 │ 2           3      0
+
+julia> df = DataFrame(cols=["a", "a", "b"], values=[1, 2, 4])
+3×2 DataFrame
+ Row │ cols    values
+     │ String  Int64
+─────┼────────────────
+   1 │ a            1
+   2 │ a            2
+   3 │ b            4
+
+julia> unstack(df, :cols, :values, allowduplicates=identity)
+1×2 DataFrame
+ Row │ a       b
+     │ Array…  Array…
+─────┼────────────────
+   1 │ [1, 2]  [4]
+
+julia> unstack(df, :cols, :values, allowduplicates=sum)
+1×2 DataFrame
+ Row │ a      b
+     │ Int64  Int64
+─────┼──────────────
+   1 │     3      4
 ```
-Note that there are some differences between the widened results above.
 """
 function unstack(df::AbstractDataFrame, rowkeys, colkey::ColumnIndex,
                  value::ColumnIndex; renamecols::Function=identity,
-                 allowmissing::Bool=false, allowduplicates::Bool=false, fill=missing)
+                 allowmissing::Bool=false,
+                 allowduplicates::Union{Bool, Function}=false, fill=missing)
+    if allowduplicates isa Function
+        groupcols = vcat(index(df)[rowkeys], index(df)[colkey])
+        @assert groupcols isa AbstractVector{Int}
+        gdf = groupby(df, groupcols)
+        if check_aggregate(allowduplicates, df[!, value]) isa AbstractAggregate
+            agg_fun = allowduplicates
+        else
+            agg_fun = Ref∘allowduplicates
+        end
+        df_agg = combine(gdf, value => agg_fun, renamecols=false)
+        group_rows = find_group_row(gdf)
+        if !issorted(group_rows)
+            df_agg = df_agg[sortperm(group_rows), :]
+        end
+        return unstack(df_agg, rowkeys, colkey, value, renamecols=renamecols,
+                    allowmissing=allowmissing, allowduplicates=false, fill=fill)
+    end
```

Comment:
Having an `else` would be more logical. Or maybe even avoid calling `return` just above and just rely on the code below doing what's needed, after `df = df_agg`?

Reply:
This is a good point - I have changed this. I have also decided to split the `operation` kwarg, as I thought it is cleaner.

Given this comment let us maybe discuss the rationale for this PR. Essentially passing `operation` is doing `combine` on `groupby` and then standard `unstack`. This is a simple thing to do on the user side. Let us confirm that we feel that indeed it is useful to add it? (I think it is useful as such pivot-table operation is quite common, but want to double check what you think)

Reply:
Yes I'd say it's useful to support this use case via `unstack` as it's realtively basic and people may look for this feature there. This is particularly useful when you have duplicates.